01 Api Versioned/src/1.Api/Api.Features/Versioned/Api.Versioned/VersionConstraint.cs

```diff
@@ -23,17 +25,22 @@ public VersionConstraint(int allowedVersion)
             IDictionary<string, object> values, HttpRouteDirection routeDirection)
         {
             if (routeDirection != HttpRouteDirection.UriResolution) return false;
-            var version = GetVersionHeader(request) ?? VersionConstants.GetSettingOrDefaultValue<int>(VersionConstants.ConfVersionDefault);
+
+            IConfigurationManagerHelper configurationManagerHelper = request.GetDependencyScope().GetService(typeof(IConfigurationManagerHelper)) as IConfigurationManagerHelper;
+            if (configurationManagerHelper == null) configurationManagerHelper = new ConfigurationManagerHelper();
+
+            var version = GetVersionHeader(request, configurationManagerHelper) ?? configurationManagerHelper.GetSettingOrDefaultValue(VersionConstants.ConfVersionDefault, VersionConstants.VersionDefault);
             return version == allowedVersion;
         }
         #endregion
 
         #region Private Methods
-        private int? GetVersionHeader(HttpRequestMessage request)
+        private int? GetVersionHeader(HttpRequestMessage request, IConfigurationManagerHelper configurationManagerHelper)
         {
             string versionAsString;
             IEnumerable<string> headerValues;
-            if (request.Headers.TryGetValues(VersionConstants.GetSettingOrDefaultValue<string>(VersionConstants.ConfVersionHeader), out headerValues) && 
+            var headerApiVersion = configurationManagerHelper.GetSettingOrDefaultValue(VersionConstants.ConfVersionHeader, VersionConstants.VersionHeader);
```

Comment:
I assume webconfig is correct. If webconfig is not corrent then the exception throws. I think it's better exception throws than an unexpected behavior.